trio/_highlevel_socket.py

```diff
@@ -114,7 +113,7 @@ def __init__(self, socket):
                         total_sent += sent
 
     async def wait_send_all_might_not_block(self):
-        async with self._send_conflict_detector:
+        with self._send_conflict_detector.sync:
```

Comment:
We can probably get rid of the async version of `ConflictDetector` entirely... that could be done here, or we could open a followup issue if it's easier.

Reply:
I considered that too but wasn't sure if there was a call for it. Have now done so.